frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts

```diff
@@ -1,8 +1,63 @@
-import { Component } from '@angular/core';
+import type { OnChanges } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import type { ExerciseSimulationBehaviorState } from 'digital-fuesim-manv-shared';
+import {
+    simulationBehaviors,
+    SimulatedRegion,
+} from 'digital-fuesim-manv-shared';
+import { ExerciseService } from 'src/app/core/exercise.service';
 
 @Component({
     selector: 'app-simulated-region-overview-behavior-tab',
     templateUrl: './simulated-region-overview-behavior-tab.component.html',
     styleUrls: ['./simulated-region-overview-behavior-tab.component.scss'],
 })
-export class SimulatedRegionOverviewBehaviorTabComponent {}
+export class SimulatedRegionOverviewBehaviorTabComponent implements OnChanges {
+    @Input() simulatedRegion!: SimulatedRegion;
+    public behaviorsToBeAdded!: ExerciseSimulationBehaviorState[];
+    public selectedBehavior?: ExerciseSimulationBehaviorState;
+    constructor(private readonly exerciseService: ExerciseService) {}
+
+    public addBehavior(behaviorState: ExerciseSimulationBehaviorState) {
+        this.exerciseService.proposeAction({
+            type: '[SimulatedRegion] Add Behavior',
+            simulatedRegionId: this.simulatedRegion.id,
+            behaviorState,
+        });
+    }
+
+    public removeBehavior() {
+        this.exerciseService.proposeAction({
+            type: '[SimulatedRegion] Remove Behavior',
+            simulatedRegionId: this.simulatedRegion.id,
+            behaviorState: this.selectedBehavior!,
+        });
+        this.selectedBehavior = undefined;
+    }
+
+    public ngOnChanges() {
+        this.calculateBehaviorsToBeAdded();
+        if (
+            this.selectedBehavior !== undefined &&
+            !this.simulatedRegion.behaviors.includes(this.selectedBehavior)
+        ) {
+            this.selectedBehavior = undefined;
+        }
+    }
+
+    public onBehaviorSelect(behavior: ExerciseSimulationBehaviorState): void {
+        this.selectedBehavior = behavior;
+    }
+
+    public calculateBehaviorsToBeAdded() {
+        this.behaviorsToBeAdded = Object.values(simulationBehaviors)
+            .filter(
+                (behavior) =>
+                    !this.simulatedRegion.behaviors.some(
+                        (behavior2) =>
+                            behavior2.type === new behavior.behaviorState().type
+                    )
+            )
+            .map((behavior) => new behavior.behaviorState());
+    }
```

Comment:
```suggestion
        this.behaviorsToBeAdded = Object.entries(simulationBehaviors)
            .filter(
                ([behaviorType]) =>
                    !this.simulatedRegion.behaviors.some(
                        (behaviorState) => behaviorState.type === behaviorType
                    )
            )
            .map(([, { behaviorState }]) => new behaviorState());
```

or (a bit less performant than above)

```suggestion
        this.behaviorsToBeAdded = Object.values(simulationBehaviors)
            .map((behavior) => new behavior.behaviorState())
            .filter(
                (behavior) =>
                    !this.simulatedRegion.behaviors.some(
                        (behaviorState) => behaviorState.type === behavior.type
                    )
            );
```

Reply:
I implemented a variant of the second approach since I don't think there is a significant performance difference and the first approach relies on the array keys being the types which is not documented or guaranteed as far as I know